src/writer.jl

```diff
@@ -73,9 +73,21 @@ end
 function _print(io::IO, pair::Pair, level::Int=0, ignore_level::Bool=false)
     key = if pair[1] === nothing
         "null" # this is what the YAML parser interprets as 'nothing'
+    elseif pair[1] isa AbstractString && (
+        occursin('#', pair[1]) ||
+            any(
+                c -> startswith(pair[1], c),
+                (
+                    '{', '}', '[', ']', '&', '*', '#', '?', '|', '-', '<', '>', '=',
+                    '!', '%', '@', ':', '`', ',', '"', '\'',
+                ),
+            )
```

Comment:
This can be written shorter as `first(pair[1]) in "{}[]..."` or the corresponding with the tuple of characters but I haven't checked if either of them is faster than this approach. Since `#` has already been checked with `occursin`, it can be removed from this list.

Reply:
Thanks! Changing to `in "{}[]..."` actually has a flat runtime on my machine, which is on average better than using `any(...)`, though slightly worse, if the match is early in the tuple. Which won't be the case most of the time, so I think using `in` is the best approach